sklearn/preprocessing/_encoders.py

```diff
@@ -571,11 +671,23 @@ def _transform_new(self, X):
 
         X_int, X_mask = self._transform(X, handle_unknown=self.handle_unknown)
 
+        if self.drop is not None:
+            max_val = max(len(cats) for cats in self.categories_)
+            to_drop = np.array([pos if pos is not None else max_val
```

Comment:
I prefer not negating if-else operator condition. `max_val if pos is None else pos`